src/main/java/org/datavyu/plugins/nativeosx/NativeOSXMediaPlayer.java

```diff
@@ -104,28 +107,29 @@ protected void playerFinish() throws MediaException { }
 
   @Override
   protected float playerGetRate() throws MediaException {
-    return nativePlayerCanvas.getRate(id);
+    return mediaPlayer.getRate(id);
   }
 
   @Override
   protected void playerSetRate(float rate) throws MediaException {
-    nativePlayerCanvas.setRate(rate, id);
+    logger.info("Setting Rate to : " + rate + "X");
+    mediaPlayer.setRate(rate, id);
   }
 
 
   @Override
   protected void playerSetStartTime(double startTime) throws MediaException {
-    playerSeek(startTime, PRECISE_SEEK_FLAG);
+    playerSeek((long) (startTime * 1000), PRECISE_SEEK_FLAG);
   }
 
   @Override
   protected double playerGetPresentationTime() throws MediaException {
-    return (nativePlayerCanvas.getCurrentTime(id) / 1000F);
+    return (mediaPlayer.getCurrentTime(id) / 1000.0);
   }
 
   @Override
   protected double playerGetFps() throws MediaException {
-    return nativePlayerCanvas.getFPS(id);
+    return mediaPlayer.getFPS(id);
```

Comment:
Refactor getFPS -> getFps

Reply:
Can't refactor this method name since it is used for the JNI call, refactoring will require rebuilding the AVFoundation project

Reply:
OK